src/runtime/mod.rs

```diff
@@ -100,4 +103,24 @@ impl AsyncRuntime {
                 .map_err(|_| ErrorKind::Io(std::io::ErrorKind::TimedOut.into()).into())
         }
     }
+
+    pub(crate) async fn resolve_address(
+        self,
+        address: &StreamAddress,
+    ) -> Result<impl Iterator<Item = SocketAddr>> {
```

Comment:
tokio's [version of the `ToSocketAddrs` trait](https://docs.rs/tokio/0.2.13/tokio/net/trait.ToSocketAddrs.html) is opaque, so you need to use `lookup_host` instead invoking trait methods directly. async-std's is basically just the same thing.